tests/baselines/reference/acceptSymbolAsWeakType.types

```diff
@@ -0,0 +1,106 @@
+=== tests/cases/compiler/acceptSymbolAsWeakType.ts ===
+const s: symbol = Symbol('s');
+>s : symbol
+>Symbol('s') : unique symbol
+>Symbol : SymbolConstructor
+>'s' : "s"
+
+const ws = new WeakSet([s]);
+>ws : WeakSet<symbol>
+>new WeakSet([s]) : WeakSet<symbol>
+>WeakSet : WeakSetConstructor
+>[s] : symbol[]
+>s : symbol
+
+ws.add(s);
+>ws.add(s) : WeakSet<symbol>
+>ws.add : (value: symbol) => WeakSet<symbol>
+>ws : WeakSet<symbol>
+>add : (value: symbol) => WeakSet<symbol>
+>s : symbol
+
+ws.has(s);
+>ws.has(s) : boolean
+>ws.has : (value: symbol) => boolean
+>ws : WeakSet<symbol>
+>has : (value: symbol) => boolean
+>s : symbol
+
+ws.delete(s);
+>ws.delete(s) : boolean
+>ws.delete : (value: symbol) => boolean
+>ws : WeakSet<symbol>
+>delete : (value: symbol) => boolean
+>s : symbol
+
+const wm = new WeakMap([[s, false]]);
+>wm : WeakMap<symbol, boolean>
+>new WeakMap([[s, false]]) : WeakMap<symbol, boolean>
+>WeakMap : WeakMapConstructor
+>[[s, false]] : [symbol, false][]
+>[s, false] : [symbol, false]
+>s : symbol
+>false : false
+
+wm.set(s, true);
+>wm.set(s, true) : WeakMap<symbol, boolean>
+>wm.set : (key: symbol, value: boolean) => WeakMap<symbol, boolean>
+>wm : WeakMap<symbol, boolean>
+>set : (key: symbol, value: boolean) => WeakMap<symbol, boolean>
+>s : symbol
+>true : true
+
+wm.has(s);
+>wm.has(s) : boolean
+>wm.has : (key: symbol) => boolean
+>wm : WeakMap<symbol, boolean>
+>has : (key: symbol) => boolean
+>s : symbol
+
+wm.get(s);
+>wm.get(s) : boolean
+>wm.get : (key: symbol) => boolean
+>wm : WeakMap<symbol, boolean>
+>get : (key: symbol) => boolean
+>s : symbol
+
+wm.delete(s);
+>wm.delete(s) : boolean
+>wm.delete : (key: symbol) => boolean
+>wm : WeakMap<symbol, boolean>
+>delete : (key: symbol) => boolean
+>s : symbol
+
+const wr = new WeakRef(s);
+>wr : WeakRef<symbol>
+>new WeakRef(s) : WeakRef<symbol>
+>WeakRef : WeakRefConstructor
+>s : symbol
+
+wr.deref();
+>wr.deref() : symbol
+>wr.deref : () => symbol
+>wr : WeakRef<symbol>
+>deref : () => symbol

```

Comment:
Hmm I was expecting `WeakRef.prototype.deref` to be `() => symbol | undefined` here. We might need to add a flag to the test so it runs in strict-mode enabling 'strictNullChecks'.  Example here: https://github.com/microsoft/TypeScript/blob/a280cafbf85df16bcd2f7258f26fdb85615b3c41/tests/cases/conformance/classes/members/privateNames/privateNameInInExpressionUnused.ts#LL1-L1C7

Reply:
That worked perfectly